src/store/index.js

```diff
@@ -21,10 +22,16 @@ export default new Vuex.Store({
     }
   },
   actions: {
-    async login ({ commit }, payload) {
-      const { data } = await axios.post(`${API_URL}/users/login`, payload)
-      commit('setCurrentUser', data);
-      router.replace('/profile')
+    login: function (state, payload) {
+      axios.post(API_URL + 'users/login', payload)
+        .then((response) => {
+          state.commit('setCurrentUser', response.data);
+          router.replace('/profile');
```

Comment:
Having Vuex call out to Vue-Router to change the URL seems really weird to me. that feels like the kind of thing the component should be handling. Feels weird to `import` the router file more than just once in the `main.js`.

Reply:
I don't completely agree here. It seems logical to just route after the request comes back. Since that happens in vuex, it seems like the routing should happen there too.
Anyway, it's a small issue so we can change it if you want.

Reply:
I'm willing to try it out in this project